source/h5vol/H5Vol_attr.c

```diff
@@ -222,16 +232,16 @@ herr_t H5VL_adios2_attr_close(void *attr, hid_t dxpl_id, void **req)
 }
 
 herr_t H5VL_adios2_attr_specific(void *obj, const H5VL_loc_params_t *loc_params,
-                                 H5VL_attr_specific_t specific_type,
-                                 hid_t dxpl_id, void **req, va_list arguments)
+                                 H5VL_attr_specific_args_t *args,
+                                 hid_t dxpl_id, void **req)
 {
     REQUIRE_NOT_NULL_ERR(obj, -1);
     H5VL_ObjDef_t *vol = (H5VL_ObjDef_t *)obj;
-    const char *attr_name = va_arg(arguments, const char *);
+    const char *attr_name =  (const char *)args->args.del.name;
```

Comment:
It won't necessarily be safe to retrieve the attribute's name here until you know that the delete operation is the one occurring since the arguments are in a union.